### DISCUSSION TITLE
Lesson 7: Revert error

### DISCUSSION THREAD

**Discussion by ayan1711** [Cyfrin/foundry-full-course-cu#799] [General]
Having this revert error in my FundMeTest.t.sol
...
Running 3 tests for test/FundMeTest.t.sol:FundMeTest
[PASS] testMinimumDollarIsFive() (gas: 5405)
[PASS] testOwnerIsMsgSender() (gas: 7733)
[FAIL. Reason: EvmError: Revert] testPriceFeedversionIsAccurate() (gas: 10157)
Traces:
  [10157] FundMeTest::testPriceFeedversionIsAccurate() 
    ├─ [5099] FundMe::getVersion() [staticcall]
    │   ├─ [0] 0x694AA1769357215DE4FAC081bf1f309aDC325306::version() [staticcall]
    │   │   └─ ← ()
    │   └─ ← "EvmError: Revert"
    └─ ← "EvmError: Revert"

Test result: FAILED. 2 passed; 1 failed; 0 skipped; finished in 6.82ms
Ran 1 test suites: 2 tests passed, 1 failed, 0 skipped (3 total tests)

...

Here is the code
...
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../src/FundMe.sol";
import {DeployFundMe} from "../script/DeployFundMe.s.sol";

contract FundMeTest is Test {
    FundMe fundMe;

    function setUp() external {
        //fundMe = new FundMe(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
    }

    function testMinimumDollarIsFive() public {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }

    function testOwnerIsMsgSender() public {
        assertEq(fundMe.i_owner(), msg.sender);
    }

    function testPriceFeedversionIsAccurate() public {
        uint256 version = fundMe.getVersion();
        assertEq(version, 4);
    }

}

...

Could anyone please help me with this :)

**Comment by Loscar1** [comment]
In your MockV3Aggregator.sol file, change the version to `4` in the line 13:

`uint256 public constant version = 4;`

**Comment by ayan1711** [comment]
Tried it but still gettin the same error :)

**Comment by Loscar1** [comment]
Please run

`forge test --mt testPriceFeedVersionIsAccurate -vvvvv --fork-url $SEPOLIA_RPC_URL`

and screenshot the log

**Reply by moejuran** [reply]
I am actually having the same problem and my version is set to 4 already.  I see there has not been a solution posted yet so I ran the command you provided and got this:

<img width="887" alt="Screenshot 2024-04-21 at 11 04 45 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/21959806/43c73c8b-a4f9-4674-8a01-941c6f7a3c73">
